src/Viewer.js

```diff
@@ -707,6 +711,32 @@ export class Viewer {
     };
   })();
 
+  /*
+  Proposed functionality for the interaction with the splats
+  The camera is the persepective camera used to render
+  The mousePosition parameter is the normalised position of the mouse
+  relative to the screen.
+
+  Uses the raycaster to traverse to the different splats to check
+  collisions with the ray and decide which one to use.
+  */
+  unprojectPositionFromSplats(renderer, camera, mousePosition) {
```

Comment:
Let's use jsdoc to document the code:

```suggestion
  /*
  * Uses the raycaster to traverse the different splats and checks for collisions.
  *
  * @param {object} camera is the perspective camera used to render
  * @param {object} position is the normalized position relative to the screen.
  * @param {object} screenSize
  * @returns {object|null} the first splat that collides with the ray.
  */
  unprojectPositionFromSplats(camera, position, screenSize) {
```

I also propose to change 
- "mousePosition" to position - it seems it doesn't matter where the position comes from
- renderer to just pass the dimensions. That's a much smaller interface that you have to pass. It also makes the method more flexible. I see that we use a similar code all over the place (for example lines 689-698)

Reply:
.. on the other hand maybe it is not a good idea to refactor other parts of the code library if we want to keep an easy rebase from upstream.

Reply:
I'm trying to get my head aroun the name:
- why "unproject"?
- maybe getSplatPosition ? - as it translates the screen position to where the splat is oriented in 3d world?